### DISCUSSION TITLE
STM32F4xx Communication via RX/TX

### DISCUSSION THREAD

**Discussion by digidiver11** [grblHAL/core#351] [General]
Hi @terjeio, i downloaded the "blackpill_map.h" file but i can't find the RX/TX pins.

Which pins are used for communication via RX/TX. In the future I would like to address my CNC machines not via USB but only via an optically separated interface, because many motor drivers cause strong interferences also during the transmission.

I myself use an optically separated current loop which manages very high data transfers (up to 10MBit).

Thanks for your great work

**Comment by terjeio** [comment]
Serial port pins are typically defined in _serial.c_ (or similarly named file). This is due to most MCUs restricting which pins can be used for which UART. You may also use the `$pins` command to output the pin mapping, but be aware that additional UART ports will not be assigned unless needed, e.g. for Modbus or Trinamic UART mode  stepper drivers.
